src/DETHRACE/common/input.c

```diff
@@ -520,20 +558,59 @@ void ChangeTextTo(int pXcoord, int pYcoord, char* pNew_str, char* pOld_str) {
     tRolling_type new_type;
     char new_char;
     LOG_TRACE("(%d, %d, \"%s\", \"%s\")", pXcoord, pYcoord, pNew_str, pOld_str);
-    NOT_IMPLEMENTED();
+
+    len = strlen(pOld_str);
+    len2 = strlen(pNew_str);
+
+    for (i = 0; i < len; i++) {
+        if (i < len2) {
+            new_char = pNew_str[i];
+        } else {
+            new_char = ' ';
+        }
+        if (new_char == ROLLING_LETTER_LOOP_RANDOM) {
+            new_type = eRT_looping_random;
+        } else if (new_char >= '0' && new_char <= '9') {
+            new_type = eRT_numeric;
+        } else {
+            new_type = eRT_alpha;
+        }
+        x_coord = gCurrent_graf_data->rolling_letter_x_pitch * i + pXcoord;
+        for (j = 0, let = gRolling_letters; j < NBR_ROLLING_LETTERS; j++, let++) {
+            if (let->number_of_letters >= 0 && let->x_coord == x_coord && let->y_coord == pYcoord) {
+                if (new_char != ROLLING_LETTER_LOOP_RANDOM) {
+                    let->letters[0] = new_char;
+                }
+                if (new_char == ' ') {
+                    let->letters[0] = ' ';
+                }
+                let->current_offset = let->number_of_letters * gCurrent_graf_data->save_slot_letter_height;
+                let->rolling_type = new_type;
+            }
+        }
+    }
+    for (i = len; i < len2; i++) {
+        // new_type should be calculated here.
```

Comment:
is this a `TODO:` or a description of the behavior?

Reply:
This comment is out of date and can be removed.

Reply:
```suggestion
```